liferay-tango/liferay-portal
# LPS-163586 Add the UI for the Preview Action to the Version History for Web Content

[Jira task](https://issues.liferay.com/browse/LPS-163586)
[Jira story](https://issues.liferay.com/browse/LPS-161020)

## Motivation

As a Content Marketer, I want to preview versions of a Web Content, so that I can easily review these versions from the Content Dashboard.

## Proposal

We are adding the barebones for the new actions dropdown for each version item.

## How to verify

~~Create some web content or document, generate several versions and visit the Content Dashboard Info Panel.~~

Currently there's no way of verify this FE implementation, as there's no exposed object from the BE. However you can simply fake props by using:

```
allVersions,
formatDate,
languageTag = 'en',
versionActions = [
		{
			action: 'preview',
			actionLabel: 'Preview',
			icon: 'view',
			previewURL:
				// eslint-disable-next-line @liferay/portal/no-localhost-reference
				'http://localhost:8080/home?p_p_id=com_liferay_journal_web_portlet_JournalPortlet&p_p_lifecycle=0&p_p_state=pop_up&p_p_mode=view&refererPlid=1&_com_liferay_journal_web_portlet_JournalPortlet_mvcPath=%2Fpreview_article_content.jsp&_com_liferay_journal_web_portlet_JournalPortlet_groupId=20121&_com_liferay_journal_web_portlet_JournalPortlet_articleId=43137&_com_liferay_journal_web_portlet_JournalPortlet_version=2.9&p_p_auth=0g9W4qxP',
			title: 'Preview',
			useModal: true,
		},
	],
```

as props for the `VersionsContent` component.

![image](https://user-images.githubusercontent.com/19485114/191252332-26ce9a46-b0cd-4f11-8618-59d8c9bf7a4d.png)
